lib/qa/authorities/crossref/generic_authority.rb

```diff
@@ -0,0 +1,59 @@
+module Qa::Authorities
+  class Crossref::GenericAuthority < Base
+    include WebServiceBase
+    class_attribute :label, :identifier
+    attr_reader :subauthority
+
+    def initialize(subauthority)
+      @subauthority = subauthority
+    end
+
+    self.label = lambda do |item|
+      [item['name'],
+       item['alt-names'].blank? ? nil : "(#{item['alt-names'].join(', ')})",
+       item['location']].compact.join(', ')
+    end
```

Comment:
I based it on the existing code in geonames (and was learning how to use lambdas) ... I've added some documentation but could easily change it to a method if you think that would be more consistent / understandable?